### ISSUE TITLE
The value of gamma is set to zero?

### ISSUE THREAD

**Issue by HqWei** [speedinghzl/CCNet#5]
Thank you very much for the source code provided. I have some questions about the parameter settings in your code. The value of gamma is set to zero?

class CrissCrossAttention(nn.Module):
    """ Pixel-wise attention module"""
    def __init__(self,in_dim):
        super(CrissCrossAttention,self).__init__()
        self.chanel_in = in_dim

        self.query_conv = nn.Conv2d(in_channels = in_dim , out_channels = in_dim//8 , kernel_size= 1)
        self.key_conv = nn.Conv2d(in_channels = in_dim , out_channels = in_dim//8 , kernel_size= 1)
        self.value_conv = nn.Conv2d(in_channels = in_dim , out_channels = in_dim , kernel_size= 1)
        self.gamma = nn.Parameter(torch.zeros(1))

    def forward(self,x):
        proj_query = self.query_conv(x)
        proj_key = self.key_conv(x)
        proj_value = self.value_conv(x)

        energy = ca_weight(proj_query, proj_key)
        attention = F.softmax(energy, 1)
        out = ca_map(attention, proj_value)
        out = self.gamma*out + x

        return out

**Comment by John1231983** [comment]
The gamma is initially set to zero and learn during training.

But I also have another question related to gamma. Why the size of gamma just `1x1`. It means each batch you only have 1 gamma, instead of `batchx1` gamma. Because during training, the batch size often bigger than 1

**Comment by Zheng222** [comment]
@John1231983 `self.gamma` can be broadcasted in pytorch.

**Comment by speedinghzl** [comment]
@John1231983 The paramter **gamma** controls how much **out** will be added into **x**, which is unrelated to the image content. So gamma has the shape **1**.

**Comment by linleon1995** [comment]
Hi, Have you ever measure the relation between gamma and input content? Is there any special observations about gamma?
